kore/src/SMT/SimpleSMT.hs

```diff
@@ -266,6 +274,26 @@ data SolverHandle = SolverHandle
     , hProc  :: !ProcessHandle
     }
 
+newtype SolverException = SolverException (Maybe ExitCode)
+    deriving (Show, Typeable)
+
+instance Exception.Exception SolverException where
+    displayException (SolverException solverExitCode) =
+        "Solver exception: The Z3 process has been unexpectedly\
+        \ terminated with exit code: "
+        <> maybe "none" show solverExitCode
+
+throwSolverException :: ProcessHandle -> IOException -> IO a
```

Comment:
Is there a reason not to catch all exceptions (`SomeException` instead of `IOException`)?

Reply:
I was thinking that this way we're fairly sure that something's wrong with accessing the solver, and it's not some unrelated exception.

Reply:
That's true; we should ignore `AsyncException`, but I think it's OK because we are re-throwing the exception anyway. Let me take care of it as part of "clean-up".